### PR TITLE
Agilex shell example

### PR DESCRIPTION
**Pull request by siclim**
Add example design with "shell" command line for intel_socfpga_agilex_socdk board. We will support shell command for all supported drivers and features for our customer through this example design.

### CODE REVIEW

**Comment by siclim** [comment]
Hi @nashif , kindly help to review this Intel SoC FPGA Agilex pull request

**Review by teburd — COMMENTED** [review]
It's nice to see a sample on a part like an agilex no less. It does seem like this could perhaps be rolled into the normal shell samples however? I didn't see anything that that precludes this from being setup as a board specific shell config. Take a look at https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/sensor/fxos8700 and how it uses board specific configs.

**Comment by siclim** [comment]
Hi @teburd,

This will be the first initial bring up patch for Agilex board specific sample code. I will introduce Agilex board specific shell command (Agilex SDM Mailbox) in my future patch, I will upstream the Mailbox driver next week.

Thanks

**Comment by siclim** [comment]
Hi @nashif @carlocaione,

Can you please help to review this pull request? Thanks a lot for your help. I don't know how to invite reviewers.

**Comment by siclim** [comment]
> > Hi @teburd,
> > This will be the first initial bring up patch for Agilex board specific sample code. I will introduce Agilex board specific shell command (Agilex SDM Mailbox) in my future patch, I will upstream the Mailbox driver next week.
> > Thanks
> 
> That makes sense, ignore my commentary above then

Thanks.

File: CODEOWNERS [hunk 1]
```diff
@@ -622,6 +622,7 @@
 /samples/basic/servo_motor/boards/*microbit* @jhe
 /samples/bluetooth/                       @jhedberg @Vudentz @alwa-nordic
 /samples/boards/intel_s1000_crb/          @sathishkuttan @dcpleung @nashif
+/samples/boards/intel_socfpga/intel_socfpga_agilex_socdk/shell/* @ngboonkhai
```
Comment (carlocaione) [inline CODEOWNERS:625]: I think `/samples/boards/intel_socfpga/intel_socfpga_agilex_socdk` is enough.

**Review by carlocaione — DISMISSED** [review]
If this will be integrate with other board specific code later LGTM

File: CODEOWNERS (hunk 1 above)
Comment (siclim) [reply CODEOWNERS]: I have updated to "/samples/boards/intel_socfpga/intel_socfpga_agilex_socdk/". Done force push. Thanks.

**Review by carlocaione — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Review by nashif — CHANGES_REQUESTED** [review]
why do we need this to be specific for a board when you can use existing and cross board shell samples? Dedicated and new samples need to do more than just that.

**Comment by siclim** [comment]
> why do we need this to be specific for a board when you can use existing and cross board shell samples? Dedicated and new samples need to do more than just that.

Hi Anas,

This will be the first initial bring up patch for Agilex board specific sample code. We will introduce Agilex board specific shell command (example: Agilex SDM Mailbox) in my near future patch to show case user the features specific to Agilex board.

Besides, "shell" is first sample designs as general purpose for all user to intel_socfpga_agilex_socdk. We also plan to include other board specific samples. So, we think of consolidate all samples for Agilex under Agilex board.

Thanks,
Siew Chin

**Comment by siclim** [comment]
Hi @nashif , is this patch acceptable to integrate into mainline?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
> This will be the first initial bring up patch for Agilex board specific sample code. We will introduce Agilex board specific shell command (example: Agilex SDM Mailbox) in my near future patch to show case user the features specific to Agilex board.

It would be better to introduce a board-level shell module instead.

**Comment by siclim** [comment]
FYI, I have also posted pull request "https://github.com/zephyrproject-rtos/zephyr/pull/43256" for Arm SIP SVC (sip_svc) driver.  Agilex SDM "mailbox" shell command will be introduced based on this sip_svc driver in future in Agilex board level shell command.

**Comment by siclim** [comment]
@nashif can this be integrated into main branch?

**Comment by nashif** [comment]
> @nashif can this be integrated into main branch?

no, we do not want to have a shell example for each board we had in the tree. The sample code does really nothing and it just enables a shell module You could probably describe how to enable the shell in the board documentation or have this sample outside of Zephyr.

**Pull request closed without merging**